ocaml/xapi/xapi_pool.ml

```diff
@@ -3717,3 +3748,44 @@ let get_guest_secureboot_readiness ~__context ~self:_ =
       `ready_no_dbx
   | _, _, _, _ ->
       `not_ready
+
+let unpack_bundle ifd =
+  let result =
+    Tar_ext.unpack_tar_file
+      ~dir:!Xapi_globs.bundle_repository_dir
+      ~ifd
+      ~max_size_limit:!Xapi_globs.bundle_max_size_limit
+  in
+  match result with
+  | Ok () ->
+      ()
+  | Error e ->
+      raise
+        Api_errors.(
+          Server_error
+            (bundle_unpacked_failed, [Tar_ext.unpack_error_to_string e])
+        )
+
+let put_bundle_handler (req : Request.t) s _ =
+  req.Request.close <- true ;
+  Xapi_http.with_context "Uploading bundle" req s (fun __context ->
+      Pool_features.assert_enabled ~__context ~f:Features.Updates ;
+      let pool = Helpers.get_pool ~__context in
+      Xapi_pool_helpers.with_pool_operation ~__context ~self:pool
+        ~doc:"pool.sync_updates" ~op:`sync_updates
+      @@ fun () ->
+      Http_svr.headers s (Http.http_200_ok ()) ;
+      let repo =
+        Repository_helpers.get_single_enabled_update_repository ~__context
+      in
+      if Db.Repository.get_origin ~__context ~self:repo = `remote then
```

Comment:
I would prefer using pattern matching here in case there would be a new case added.